mlx/traceability.py

```diff
@@ -134,51 +134,45 @@ def run(self):
             caption = self.arguments[1].replace('\n', ' ')
 
         # Store item info
-        if targetid in env.traceability_all_items and env.traceability_all_items[targetid]['placeholder'] is False:
```

Comment:
It's just this and the 6 lines below that were removed? Could you explain a bit?

Reply:
Sphinx doesn't parse all .rst files when doing a re-build (probably comparable how a makefile only builds the changed c-files for an executable).

Change at the bottom of the file:
Old solution started from a clean database of items, so when sphinx decided to not re-build some rst-files, the items in these rst-files would be missing in the database. This lead to warnings about missing items (only for re-builds). New solution re-uses the database from a previous build: only create the database when it is not there yet.

Change at the top of the file:
Since all items of a previous build are already in the database, we can't know if the currently parsed item is already in the database because:
- a) it was parsed on the previous build and still is in the database, or
- b) it is duplicate in the rst-sources.
I logged issue #12 for this, and we should solve it soon...

Note: we never saw it in our private project(s) as we stage all of our rst-files and build from there. Since we stage on every re-build i think sphinx detects this and parses all rst-files again and again on every re-build...